diff --git a/package.json b/package.json
index a7261ff..f4fa152 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "current",
+  "name": "upgrade",
   "version": "0.1.0",
   "private": true,
   "scripts": {
@@ -33,11 +33,11 @@
     "eslint-config-next": "^13.2.4",
     "postcss": "^8.4.21",
     "prettier": "^2.8.6",
-    "prettier-plugin-tailwindcss": "^0.2.5",
-    "tailwindcss": "^3.2.7",
+    "prettier-plugin-tailwindcss": "^0.2.6",
+    "tailwindcss": "^3.3.0",
     "typescript": "^5.0.2"
   },
   "ct3aMetadata": {
-    "initVersion": "7.9.0"
+    "initVersion": "7.10.1"
   }
 }
diff --git a/src/server/api/trpc.ts b/src/server/api/trpc.ts
index defe423..5d1335a 100644
--- a/src/server/api/trpc.ts
+++ b/src/server/api/trpc.ts
@@ -27,7 +27,7 @@ type CreateContextOptions = Record<string, never>;
  * - testing, so we don't have to mock Next.js' req/res
  * - tRPC's `createSSGHelpers`, where we don't have req/res
  *
- * @see https://create.t3.gg/en/usage/trpc#-servertrpccontextts
+ * @see https://create.t3.gg/en/usage/trpc#-serverapitrpcts
  */
 const createInnerTRPCContext = (_opts: CreateContextOptions) => {
   return {};
diff --git a/tailwind.config.cjs b/tailwind.config.ts
similarity index 50%
rename from tailwind.config.cjs
rename to tailwind.config.ts
index a82e7e9..d4d3fa2 100644
--- a/tailwind.config.cjs
+++ b/tailwind.config.ts
@@ -1,10 +1,9 @@
-/** @type {import('tailwindcss').Config} */
-const config = {
+import { type Config } from "tailwindcss";
+
+export default {
   content: ["./src/**/*.{js,ts,jsx,tsx}"],
   theme: {
     extend: {},
   },
   plugins: [],
-};
-
-module.exports = config;
+} satisfies Config;
